dvc/updater.py

```diff
@@ -40,36 +38,80 @@ def check(self):
         Logger.info('Checking for updates...')
 
         try:
-            r = requests.get(self.URL, timeout=self.TIMEOUT_GET)
-            j = r.json()
-            latest = j['version']
-            open(self.updater_file, 'w+').close()
+            self._get_latest_version()
         except Exception as exc:
             msg = 'Failed to obtain latest version: {}'.format(str(exc))
             Logger.debug(msg)
             return
 
-        l_major, l_minor, l_patch = [int(x) for x in latest.split('.')]
-        c_major, c_minor, c_patch = [int(x) for x in current.split('.')]
+        if self._is_outdated():
+            self._notify()
 
-        if l_major <= c_major and \
-           l_minor <= c_minor and \
-           l_patch <= c_patch:
-            return
+    def _get_latest_version(self):
+        r = requests.get(self.URL, timeout=self.TIMEOUT_GET)
+        j = r.json()
+        self.latest = j['version']
+        open(self.updater_file, 'w+').close()
+
+    def _is_outdated(self):
+        l_major, l_minor, l_patch = [int(x) for x in self.latest.split('.')]
+        c_major, c_minor, c_patch = [int(x) for x in self.current.split('.')]
 
-        msg = (
+        return (l_major > c_major or
+                l_minor > c_minor or
+                l_patch > c_patch)
+
+    def _notify(self):
+        message = (
             'Update available {red}{current}{reset} -> {green}{latest}{reset}'
-            '\n'
-            'Run {yellow}pip{reset} install dvc {blue}--upgrade{reset}'
-            .format(
-                red=colorama.Fore.RED,
-                reset=colorama.Fore.RESET,
-                green=colorama.Fore.GREEN,
-                yellow=colorama.Fore.YELLOW,
-                blue=colorama.Fore.BLUE,
-                current=current,
-                latest=latest,
+            + '\n'
+            + self._get_update_instructions()
+        ).format(red=colorama.Fore.RED,
+                 reset=colorama.Fore.RESET,
+                 green=colorama.Fore.GREEN,
+                 yellow=colorama.Fore.YELLOW,
+                 blue=colorama.Fore.BLUE,
+                 current=self.current,
+                 latest=self.latest)
+
+        if sys.stdout.isatty():
+            Logger.box(message, border_color='yellow')
+
+    def _get_update_instructions(self):
```

Comment:
CodeClimate sadly things that my code is too complex for the simple mind, because every `if` `else` adds a level of complexity; I can have a dictionary with the operative systems and their package manager, and then another dicitonary with the package manager and their respective messages, reducing the _cognitive complexity_.

However, reading the following is somehow more complex than a "switch case":
```python
messages[package_manager[distro.id()]]
```

Reply:
Yes CC is sometimes too strict. It's fine sometimes to mark them as `Won't fix` and move on.